public/source/server.js

```diff
@@ -96,7 +97,18 @@ Server.prototype._getCredentials = function(callback) {
 Server.prototype.watchRepository = function(repositoryPath, callback) {
   this.socket.emit('watch', { path: repositoryPath }, callback);
 };
-
+Server.prototype.queryPromise = function(type, url, arg) {
```

Comment:
Maybe implement
`getPromise`
`postPromise`
`delPromise`
or make `query` to return a promise with backward compatibility for `callback`?